CMakeLists.txt

```
@@ -252,10 +252,9 @@ if(ENABLE_GUI OR ENABLE_CLI)
         list(APPEND PNG_PNG_INCLUDE_DIR ${CMAKE_BINARY_DIR}/extlib/libpng)
 
         find_vendored_package(Freetype freetype
-            WITH_ZLIB               OFF
-            WITH_BZip2              OFF
-            WITH_PNG                OFF
-            WITH_HarfBuzz           OFF
+            FT_DISABLE_BZIP2        ON
+            FT_DISABLE_HARFBUZZ     ON
+            FT_DISABLE_BROTLI       ON
```

```suggestion
            FT_DISABLE_BROTLI       ON
            FT_DISABLE_PNG          ON
            FT_DISABLE_ZLIB         ON
```

We should disable PNG - as it was before.
And I do not see why not do `FT_DISABLE_ZLIB=ON` - it disables the system ZLIB and uses the internal one anyway.

---

I'm fine with disabling libpng (which disables support of PNG compressed OpenType embedded bitmaps), although we build it anyway as a direct dependency of SolveSpace, so I'm not sure why we would want to prevent FreeType from using it.

As for zlib, "system" is a bit confusing here. When disabling zlib it does not use the zlib that we build as a direct dependency of SolveSpace, but rather the copy of zlib vendored in FreeType, so we just end up with two copies of zlib in the binary, each from a different version. Without disabling it, CMake ends up using the SolveSpace zlib that we introduce into the build before adding FreeType, so we have only a single version of zlib in the binary.

---

> so I'm not sure why we would want to prevent FreeType from using it.

I'm not sure what "PNG compressed OpenType embedded bitmaps." are but I presume that since they are bitmaps they would be useless for the text feature of SolveSpace, which uses only the curves from the fonts.

However enabling the option causes the EXE (Windows 32 bit build with MSVC 2019 with LTO) size to **_decrease_** for me:

FT_DISABLE_PNG          ON:
5.93 MB (6,218,752 bytes)

FT_DISABLE_PNG          OFF:
5.92 MB (6,208,512 bytes)

WTF?!

The `FT_DISABLE_ZLIB` option does **_not_** influence the EXE size. So let us leave them as you configured them.

---

Weird. On Linux I see the reverse (when building with vendored zlib, libpng, and freetype):

```
$ ls -ln --time-style=+"" bin/solvespace
-rwxr-xr-x 1 1000 1000 52871080  bin/solvespace
$ ls -ln --time-style=+"" ../solvespace-png
-rwxr-xr-x 1 1000 1000 52884800  ../solvespace-png
```

I'll try to boot into Windows and check what's happening there when I have some time.

---

Same result, though to a bit lesser effect, with `-DCMAKE_BUILD_TYPE=Release`:

```
$ ls -ln --time-style=+"" bin/solvespace
-rwxr-xr-x 1 1000 1000 5290248  bin/solvespace
$ ls -ln --time-style=+"" ../solvespace-png
-rwxr-xr-x 1 1000 1000 5294640  ../solvespace-png
```

---

@ruevs I'm having trouble reproducing your results on Windows (Visual Studio 2022, with `-DCMAKE_BUILD_TYPE=Release-DENABLE_LTO=1`):

```
>dir /c bin\solvespace.exe ..\solvespace-png.exe
8,433,152 solvespace.exe
               1 File(s)      8,433,152 bytes

8,436,224 solvespace-png.exe
               1 File(s)      8,436,224 bytes
```

Given the fact that as you noted embedded bitmaps support is unneeded, and that on both Windows and Linux I see an increase in file size of 3-4KB, it's probably best if we disable libpng support in FreeType. Should I submit a PR?

---

Done.